proposals/NNNN-allow-let-property-wrapper.md

```diff
@@ -0,0 +1,239 @@
+}
+```
+Permitting wrapped properties to mimic the rules for other type instances that can be written with either a `var` or a `let` will simplify the Swift language. Additionally, `let` wrapped properties would expand property wrapper usage to allow `nonisolated` wrapped properties and simplify property wrappers usage where the backing type is a reference type.  
+
+
+## Proposed solution
+
+We propose to allow the application of property wrappers to let declared properties, which will permit the wrapper type to be initialized only once without affecting the implementation of the backing storage.
+
+For example, the following is the implementation for a `@BoilingPoint` property wrapper:
+```swift
+@propertyWrapper
+struct BoilingPoint<T> {
+  init(wrappedValue: T) {
+    self.wrappedValue = wrappedValue
+  }
+
+  var wrappedValue: T
+  var projectedValue: T
+  
+  func toFarenheit() {}
+}
+
+struct Temperature {
+  @BoilingPoint var water: Double = 373.1
+}
+```
+
+The initial value for `water` is set using the `init(wrappedValue:)` initializer of the `BoilingPoint` property wrapper type. To ensure that `water` is not changed again, we could add logic to the `wrappedValue` setter to check if the property was already initialized and prevent re-assignment. However, this is an inconvenient solution.
+
+Instead, we could declare the `water` property with a `let`, synthesize a local `let` constant for the backing storage property (prefixed with an underscore), and only allow the property wrapper to be initialized once, passing the assigned value to `init(wrappedValue:)`. Now, rewriting `Temperature`’s properties as `let` constants will translate to:
+```swift
+struct Temperature {
+  @BoilingPoint let water: Double = 373.1
+
+  // ... introduces _water stored property
+  private let _water: BoilingPoint<Double> = BoilingPoint<Double>(wrappedValue: 373.1)
+  
+  // ... getter-only computed property
+  @BoilingPoint var water: Double {
+    get { return self._water.wrappedValue }
+  }
+  
+  // ... getter-only projectedValue property
+  internal var $water: Double {
+    get { return self._water.projectedValue }
+  }
+}
+```
+
+Declaring a `let` wrapped property of `water` of type `BoilingPoint` generates a `let` declared storage property of `_water`, a getter-only computed property that assigns the declared value to the backing property wrapper's `wrappedValue`. A let declared wrapped property also generates `$water`, a getter-only projectedValue property.
+
+The `let` declared property can also be initialized once via the generated storage property. For example, reusing the above example:
+```swift
+  @BoilingPoint let water: Double
+  
+  init() {
+    _water = BoilingPoint(wrappedValue: 373.1)
+  }
+```
+
+After the `let` declared property has been initialized, attempting to reassign it will show an error. Clients can continue to manipulate the projection just like with `var` declared properties.
```

Comment:
```suggestion
After the `let` declared property has been initialized, attempting to reassign it will be an error. Users can continue to manipulate the projection just as with `var` declared properties:

```